redisinsight/ui/src/pages/vector-search/saved-queries/SavedQueriesScreen.tsx

```diff
@@ -62,35 +122,35 @@ export const SavedQueriesScreen = ({
       </VectorSearchScreenHeader>
       <VectorSearchScreenFooter grow={1} padding={6}>
         <VectorSearchSavedQueriesContentWrapper>
-          <VectorSearchSavedQueriesSelectWrapper>
-            <Title size="S">Index:</Title>
-            <RiSelect
-              loading={false}
-              disabled={false}
-              options={savedIndexes}
-              value={selectedIndex?.value}
-              data-testid="select-saved-index"
-              onChange={onIndexChange}
-              valueRender={({ option, isOptionValue }) =>
-                isOptionValue ? (
-                  option.value
-                ) : (
-                  <TagsWrapper>
-                    {option.value}
-                    {option.tags.map((tag) => (
-                      <FieldTag key={tag} tag={tag as any} />
-                    ))}
-                  </TagsWrapper>
-                )
-              }
-            />
-          </VectorSearchSavedQueriesSelectWrapper>
-          {selectedIndex?.queries.map((query) => (
-            <VectorSearchScreenBlockWrapper
-              key={query.value}
-              // as="div"
-              padding={6}
-            >
+          {savedIndexes.length > 0 ? (
+            <VectorSearchSavedQueriesSelectWrapper>
+              <Title size="S">Index:</Title>
+              <RiSelect
+                loading={false}
+                disabled={false}
+                options={savedIndexes}
+                value={selectedIndexItem?.value}
+                data-testid="select-saved-index"
+                onChange={onIndexChange}
+                valueRender={({ option, isOptionValue }) =>
+                  isOptionValue ? (
+                    option.value
+                  ) : (
+                    <TagsWrapper>
+                      {option.value}
+                      {option.tags.map((tag) => (
+                        <FieldTag key={tag} tag={tag as any} />
```

Comment:
nit: Can't we use `as FieldTypes` instead of `as any`, or do we need to break the type checking for some reason?